dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/CompanyUploader.kt

```diff
@@ -15,34 +18,53 @@ import java.net.SocketTimeoutException
 @Service
 class CompanyUploader(
     @Autowired private val companyDataControllerApi: CompanyDataControllerApi,
+    @Autowired private val objectMapper: ObjectMapper,
 ) {
     companion object {
         const val MAX_RETRIES = 3
         const val UNAUTHORIZED_CODE = 401
-        const val FORBIDDEN_CODE = 403
     }
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
-    /**
-     * Uploads a single Company to the dataland backend performing at most MAX_RETRIES retries.
-     * This function absorbs errors and logs them.
-     */
-    fun uploadSingleCompany(
-        companyInformation: CompanyInformation,
-    ) {
+    @Suppress("ReturnCount")
+    private fun checkForDuplicateIdentifierAndGetConflictingCompanyId(exception: ClientException): String? {
+        if (exception.statusCode != HttpStatus.BAD_REQUEST.value()) {
+            return null
+        }
+
+        val exceptionResponse = exception.response!! as ClientError<*>
+        val exceptionBodyString = exceptionResponse.body.toString()
+
+        val errorResponseBody = objectMapper.readTree(exceptionBodyString)
+        val firstError = errorResponseBody.get("errors").get(0)
+        if (firstError.get("errorType").textValue() != "duplicate-company-identifier") {
+            return null
+        }
+
+        val conflictingIdentifiers = firstError.get("metaInformation")
+        if (!conflictingIdentifiers.isArray || conflictingIdentifiers.size() != 1) {
+            return null
+        }
+
+        val conflictingIdentifier = conflictingIdentifiers.get(0)
+        val conflictingIdentifierType = conflictingIdentifier.get("identifierType").textValue()
+        if (conflictingIdentifierType != "Lei") {
```

Comment:
Why filter this.

Reply:
IMO any input that gets returned by a network request should be considered untrusted and validated before it is being used. Checking if the returned object has a field "identifierType" of value "Lei" is just one of the sanity checks.

Reply:
Of course. Using `"Lei"` as a string botheres me a bit. But since this is only an additional check, I will overcome that. We can move the corresponding Enum to the backend-utils or maybe a more company specific library later.

Reply:
I saw that the `IdentifierType` enum is actually part of the backend clients. So let us use it here.

Reply:
Fair point. Changed.